chainer/links/connection/linear.py

```diff
@@ -120,5 +120,8 @@ def __call__(self, x):
 
         """
         if self.W.data is None:
-            self._initialize_params(x.size // x.shape[0])
+            in_size = 1
```

Comment:
How about `functools.reduce(operator.mul, x.shape[1:], 1)`? It's faster and easier to read than manual multiplication.

Reply:
Using functools and operator now.